### PR TITLE
Add http proxy support

### PR DESCRIPTION
**Pull request by alexjfisher**
This is a fairly trivial change to allow an `http_proxy` to be specified.

### CODE REVIEW

File: lib/puppet/reports/slack.rb [hunk 1]
```diff
@@ -94,7 +99,7 @@ def post_to_webhook(uri, payload)
     when Net::HTTPSuccess
       return
     else
-      Puppet.err("Notification sent faild to slack channel: #{channel}")
```
Comment (alexjfisher) [inline lib/puppet/reports/slack.rb:97]: There was a bug here. `channel` isn't in scope.

**Comment by alexjfisher** [comment]
@udzura Does this look ok, or are they any changes you would like me to make?  Thanks.

**Comment by LightAxe** [comment]
I definitely need http_proxy support; is this merged in yet?

**Comment by udzura** [comment]
Thanks!! sorry for late.

**Pull request merged**